ledger/ledger-common/src/main/java/net/corda/v5/ledger/common/NotaryLookup.java

```diff
@@ -0,0 +1,47 @@
+package net.corda.v5.ledger.common;
+
+import net.corda.v5.base.annotations.DoNotImplement;
+import net.corda.v5.base.annotations.Suspendable;
+import net.corda.v5.base.types.MemberX500Name;
+import net.corda.v5.membership.NotaryInfo;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+
+/**
+ * Defines a mechanism to allow flows to retrieve the NotaryInfo in the network.
+ * The platform will provide an instance of NotaryLookup to flows via property injection.
+ */
+@DoNotImplement
+public interface NotaryLookup {
+
+    /**
+     * Gets a {@link Collection} of {@link NotaryInfo} services available on the network.
+     *
+     * @return Returns a {@link Collection} of {@link NotaryInfo} services available on the network.
+     */
+    @NotNull
+    @Suspendable
+    // TODO : Consider List<NotaryInfo>
+    Collection<NotaryInfo> getNotaryServices();
+
+    /**
+     * Looks up the notary information of a notary by legal name.
+     *
+     * @param notaryServiceName The {@link MemberX500Name} of the notary service to retrieve.
+     * @return Returns the {@link NotaryInfo} that matches the specified notary service name, or null if no such notary exists.
+     */
+    @Nullable
+    @Suspendable
+    NotaryInfo lookup(@NotNull MemberX500Name notaryServiceName);
```

Comment:
Not specific to this code - but should we consider using `Optional` rather than returning null values, to make it more explicit on the interface?

Reply:
Having discussed with @relyafi (https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html) this might be preferable to returning nullable values, but we might need to consider:

 - They might not be serializable
 - We'd really need to be consistent across the codebase, and upgrade all nullable returns to optional
 
 and we suspect that owing to deadlines, this might not be possible right now.